docs/api.md

```diff
@@ -656,29 +671,64 @@ TODO(pavelkalinnikov): Consider renaming it to AddLogLeafResult or the like.
 <a name="trillian.TrillianLog"></a>
 
 ### TrillianLog
-Provides access to a Verifiable Log data structure as defined in the
-[Verifiable Data Structures](docs/papers/VerifiableDataStructures.pdf) paper.
-
-The API supports adding new entries to be integrated into the log&#39;s tree. It
-does not provide arbitrary tree modifications. Additionally, it has read
-operations such as obtaining tree leaves, inclusion/consistency proofs etc.
+The TrillianLog service provides access to an append-only Log data structure
+as described in the [Verifiable Data
+Structures](docs/papers/VerifiableDataStructures.pdf) paper.
+
+The API supports adding new entries to the Merkle tree for a specific Log
+instance (identified by its log_id) in two modes:
+ - For a normal log, new leaf entries are queued up for subsequent
+   inclusion in the log, and the leaves are assigned consecutive leaf_index
+   values as part of that integration process.
+ - For a &#39;pre-ordered log&#39;, new entries have an already-defined leaf
+   ordering, and leaves are only integrated into the Merkle tree when a
+   contiguous range of leaves is available.
+
+The API also supports read operations to retrieve leaf contents, and to
```

Comment:
proofs of inclusion and the append-only...?

Reply:
Done.